### PR TITLE
chore: Remove WithImplicitStyles and M3 prefix

### PR DESCRIPTION
**Pull request by kazo0**
Remove the WithImplicitStyles property for MaterialResourcesV2

Remove M3 Prefix for Uno.Material v2 styles

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
Running this build fails with:

```
**System.ArgumentException:** 'Missing resource: key=MaterialHyperlinkButtonStyle'
```

Likely because of the hyperlink button change. Let requeue a build.

File: src/samples/Uno.Themes.Samples/Uno.Themes.Samples.Shared/Content/NestedSamples/M3MaterialCommandBarSample_NestedPage1.xaml [hunk 1]
```diff
@@ -47,8 +47,8 @@
 					Tag="See code behind"
 					Style="{StaticResource SeeSourceFlyoutButtonStyle}" />
 
-			<TextBlock Style="{StaticResource M3MaterialBodySmall}">
-				note: In order for the `M3MaterialCommandBarStyle` to work properly on iOS and Android,
+			<TextBlock Style="{StaticResource MaterialBodySmall}">
```
Comment (agneszitte-nventive) [inline src/samples/Uno.Themes.Samples/Uno.Themes.Samples.Shared/Content/NestedSamples/M3MaterialCommandBarSample_NestedPage1.xaml:50]: Should be Body small without material here no?
Same for below

**Review by agneszitte-nventive — APPROVED** [review]

File: src/library/Uno.Material/MaterialResourcesV2.cs [hunk 2]
```diff
@@ -77,66 +69,66 @@ private IEnumerable<(string ResourceKey, string SharedKey, bool IsDefaultStyle)>
...
-			Add("M3MaterialSecondaryLargeFabStyle");
-			Add("M3MaterialTertiaryLargeFabStyle");
-			Add("M3MaterialExtendedFabStyle");
-			Add("M3MaterialSurfaceExtendedFabStyle");	
-			Add("M3MaterialSecondaryExtendedFabStyle");	
-			Add("M3MaterialTertiaryExtendedFabStyle");
-			Add("M3MaterialSliderStyle");
-			Add("M3MaterialHyperlinkButtonStyle", isImplicit: true);
-			Add("M3MaterialSecondaryHyperlinkButtonStyle");
+			Add("MaterialCheckBoxStyle", isImplicit: true);
```
Comment (agneszitte-nventive) [inline src/library/Uno.Material/MaterialResourcesV2.cs:72]: Is the related doc for the matrix style needs also some update ? Missing styles or implicit column?

**Review by agneszitte-nventive — APPROVED** [review]

**Comment by jeromelaban** [comment]
```
**System.ArgumentException:** 'Missing resource: key=MaterialHyperlinkButtonStyle'
```

It still there after the rebuild.

File: src/library/Uno.Material/MaterialResourcesV2.cs (hunk 2 above)
Comment (Xiaoy312) [reply src/library/Uno.Material/MaterialResourcesV2.cs:72]: you can ran the script again; it will parse this method && all styles in xaml
to do a join

**Review by agneszitte-nventive — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

File: src/library/Uno.Material/Styles/Controls/v2/CalendarDatePicker.xaml [hunk 3]
```diff
@@ -208,14 +208,14 @@
 							<TextBlock HorizontalAlignment="Stretch"
 									   VerticalAlignment="Top"
 									   Text="{TemplateBinding Header}"
-									   Style="{StaticResource M3MaterialBodySmall}"
+									   Style="{StaticResource BodySmall}"
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/CalendarDatePicker.xaml:211]: use the full style name here, alias should not be used here

Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/CalendarDatePicker.xaml:211]: same in CalendarView + DatePicker too

Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/CalendarDatePicker.xaml:211]: there is issue with "LabelLarge, LabelMedium" and probably all other typography style everywhere
use the script to check for references

File: src/library/Uno.Material/MaterialResourcesV2.cs [hunk 4]
```diff
@@ -15,28 +15,26 @@ namespace Uno.Material
...
-
+		private const string StylePrefix = "Material";
 		public MaterialResourcesV2()
 		{
 			Source = new Uri("ms-appx:///Uno.Material/Generated/mergedpages.v2.xaml");
 
 			MapStyleInfo();
 		}
 
-		public bool WithImplicitStyles { set => ExportImplicitStyles(value); }
```
[the remark this answers is not in the document]
Comment (kazo0) [reply src/library/Uno.Material/MaterialResourcesV2.cs:28]: It's updated in the Material Control Styles matrix page, is there someplace else that I'm missing?

**Review by Xiaoy312 — CHANGES_REQUESTED** [review]
see last review

File: src/library/Uno.Material/MaterialResourcesV2.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Xiaoy312) [reply src/library/Uno.Material/MaterialResourcesV2.cs:28]: sry missed that

**Review by Xiaoy312 — COMMENTED** [review]
(resetting vote)

**Review by Xiaoy312 — APPROVED** [review]
(resetting vote)

**Pull request merged**